### PR TITLE
152502- Added different profiles for Performance Test rerun 

### PR DESCRIPTION
**Pull request by MitaliSingh16**
-  PR raised for uploading performance test script for ERP Facade-Webhook endpoint with Different profiles .
  - Profile1:  3 events per second for 30 minutes
  - Profile2:  4 events per second for 30 minutes
  - Profile3:  2 events per second for 30 minutes
- The script is capturing and reporting  performance metrics along with Performance HTML report.
- Dependency Checker Vulnerability issue fixed.

### CODE REVIEW

File: Performance Script/PayloadDataSetup/PayloadSetupDifferentProfiles.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    var productArraySize = payload.data.products.length;
+    var unitOfSalesArraySize = payload.data.unitsOfSale.length;
+    var jsonObj = payload;
+
+    payload.data.correlationId = generateUniqueCorrId();
+
+    for (var i = 0; i < productArraySize; i++) {
+
+        jsonObj.data.products[i].permit = Config.Permit;
+
```
Comment (pravinvarade10801) [inline Performance Script/PayloadDataSetup/PayloadSetupDifferentProfiles.js:14]: Can we remove this extra blank line

File: Performance Script/WebhookLoadTestLoadTestWith2rps.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+};
+
+export function setup() {
+    const event = new Date(Date.now());
+    console.log("start time:" + event.toUTCString());
+
+}
+
+
```
Comment (pravinvarade10801) [inline Performance Script/WebhookLoadTestLoadTestWith2rps.js:34]: Can we remove extra blank line

File: Performance Script/WebhookLoadTestLoadTestWith2rps.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    console.log(updatedPayload1.data.correlationId);
+
+    check(res, {
+        'Status is 200': (r) => r.status === 200,
+    });
+
+    console.log("Status code:" + res.status);
+
+}
+export function ScenarioWithTwoProduct() {
```
Comment (pravinvarade10801) [inline Performance Script/WebhookLoadTestLoadTestWith2rps.js:94]: We should add one blank line between two functions

File: Performance Script/WebhookLoadTestLoadTestWith2rps.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    console.log("In ScenarioWithTwoProduct");
+    console.log(updatedPayload2.data.correlationId);
+
+    check(res, {
+        'Status is 200': (r) => r.status === 200,
+    });
+
+    console.log("Status code:" + res.status);
+}
+export function ScenarioWithHundredProduct() {
```
Comment (pravinvarade10801) [inline Performance Script/WebhookLoadTestLoadTestWith2rps.js:107]: Please add one blank line

File: Performance Script/WebhookLoadTestLoadTestWith2rps.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    const res = http.post(url.toString(), JSON.stringify(updatedPayload1), { headers }, { tags: { my_custom_tag: 'ScenarioWithOneProduct' } });
+    console.log("In ScenarioWithOneProduct:");
+    console.log(updatedPayload1.data.correlationId);
+
+    check(res, {
+        'Status is 200': (r) => r.status === 200,
+    });
+
+    console.log("Status code:" + res.status);
+
```
Comment (pravinvarade10801) [inline Performance Script/WebhookLoadTestLoadTestWith2rps.js:92]: Please remove extra blank line

**Review by pravinvarade10801 — COMMENTED** [review]
Have added some minor comments.

File: Performance Script/PayloadDataSetup/PayloadSetupDifferentProfiles.js (hunk 1 above)
Comment (MitaliSingh16) [reply Performance Script/PayloadDataSetup/PayloadSetupDifferentProfiles.js:14]: done

File: Performance Script/WebhookLoadTestLoadTestWith2rps.js (hunk 2 above)
Comment (MitaliSingh16) [reply Performance Script/WebhookLoadTestLoadTestWith2rps.js:34]: done

File: Performance Script/WebhookLoadTestLoadTestWith2rps.js (hunk 3 above)
Comment (MitaliSingh16) [reply Performance Script/WebhookLoadTestLoadTestWith2rps.js:94]: done

File: Performance Script/WebhookLoadTestLoadTestWith3rps.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    const res = http.post(url.toString(), JSON.stringify(updatedPayload1), { headers }, { tags: { my_custom_tag: 'ScenarioWithOneProduct' } });
+    console.log("In ScenarioWithOneProduct");
+    console.log(updatedPayload1.data.correlationId);
+
+    check(res, {
+        'Status is 200': (r) => r.status === 200,
+    });
+
+    console.log("Status code:" + res.status);
+
```
[the remark this answers is not in the document]
Comment (MitaliSingh16) [reply Performance Script/WebhookLoadTestLoadTestWith3rps.js:90]: done

File: Performance Script/WebhookLoadTestLoadTestWith2rps.js (hunk 5 above)
Comment (MitaliSingh16) [reply Performance Script/WebhookLoadTestLoadTestWith2rps.js:92]: done

File: Performance Script/WebhookLoadTestLoadTestWith4rps.js [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    const res = http.post(url.toString(), JSON.stringify(updatedPayload1), { headers }, { tags: { my_custom_tag: 'ScenarioWithOneProduct' } });
+    console.log("In ScenarioWithOneProduct");
+    console.log(updatedPayload1.data.correlationId);
+
+    check(res, {
+        'Status is 200': (r) => r.status === 200,
+    });
+
+    console.log("Status code:" + res.status);
+
```
[the remark this answers is not in the document]
Comment (MitaliSingh16) [reply Performance Script/WebhookLoadTestLoadTestWith4rps.js:91]: done

File: Performance Script/WebhookLoadTestLoadTestWith4rps.js [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+    console.log("In ScenarioWithTwoProduct");
+    console.log(updatedPayload3.data.correlationId);
+
+    check(res, {
+        'Status is 200': (r) => r.status === 200,
+    });
+
+    console.log("Status code:" + res.status);
+}
+export function ScenarioWithHundredProduct() {
```
[the remark this answers is not in the document]
Comment (MitaliSingh16) [reply Performance Script/WebhookLoadTestLoadTestWith4rps.js:106]: done

File: Performance Script/WebhookLoadTestLoadTestWith3rps.js [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+    console.log("In ScenarioWithTwoProduct");
+    console.log(updatedPayload2.data.correlationId);
+
+    check(res, {
+        'Status is 200': (r) => r.status === 200,
+    });
+
+    console.log("Status code:" + res.status);
+}
+export function ScenarioWithHundredProduct() {
```
[the remark this answers is not in the document]
Comment (MitaliSingh16) [reply Performance Script/WebhookLoadTestLoadTestWith3rps.js:102]: Done

**Review by pravinvarade10801 — APPROVED** [review]
LGTM

File: Performance Script/WebhookLoadTestLoadTestWith3rps.js [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+import { htmlReport } from "https://raw.githubusercontent.com/benc-uk/k6-reporter/main/dist/bundle.js";
+import { textSummary } from "https://jslib.k6.io/k6-summary/0.0.1/index.js";
+import { URL } from 'https://jslib.k6.io/url/1.0.0/index.js';
+import { PayloadSetup } from './PayloadDataSetup/PayloadSetupDifferentProfiles.js';
+
+
+var Config = JSON.parse(open('./config.json'));
+var ProductList = JSON.parse(open('./PayloadData/WebhookPayloads/SAPProductList.json'));
+
+var defaultPayload1 = JSON.parse(open('./PayloadData/WebhookPayloads/1ProductNewCell.json'));
```
Comment (Vishal14583) [inline Performance Script/WebhookLoadTestLoadTestWith3rps.js:13]: Can we rename these variables please.

File: Performance Script/WebhookLoadTestLoadTestWith3rps.js [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+            executor: 'constant-arrival-rate',
+            exec: 'ScenarioWithOneProduct',
+            rate: 3,
+            timeUnit: '1s',
+            startTime: '1s',
+            duration: '27m',
+            preAllocatedVUs: 5,
+            maxVUs: 12,
+        },
+        ScenarioWithTwoProduct: {
```
Comment (Vishal14583) [inline Performance Script/WebhookLoadTestLoadTestWith3rps.js:55]:
```suggestion
        ScenarioWithTwoProducts: { 
```

File: Performance Script/WebhookLoadTestLoadTestWith3rps.js [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+            executor: 'constant-arrival-rate',
+            exec: 'ScenarioWithTwoProduct',
+            rate: 3,
+            timeUnit: '1s',
+            startTime: '27m',
+            duration: '179s',
+            preAllocatedVUs: 5,
+            maxVUs: 12
+        },
+        ScenarioWithHundredProduct: {
```
Comment (Vishal14583) [inline Performance Script/WebhookLoadTestLoadTestWith3rps.js:65]:
```suggestion
        ScenarioWithHundredProducts: { 
```

Comment (Vishal14583) [reply Performance Script/WebhookLoadTestLoadTestWith3rps.js:65]: Apply this wherever applicable.

File: Performance Script/WebhookLoadTestLoadTestWith3rps.js [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+            startTime: '1799s',
+            duration: '1s',
+            preAllocatedVUs: 1
+        }
+    }
+};
+
+export function ScenarioWithOneProduct() {
+
+    const updatedPayload1 = PayloadSetup(defaultPayload1, ProductList);
```
Comment (Vishal14583) [inline Performance Script/WebhookLoadTestLoadTestWith3rps.js:79]: Rename please

**Review by Vishal14583 — CHANGES_REQUESTED** [review]
Added few review comments. Please apply the fix wherever applicable across the script.

**Review by Sadha1501493 — CHANGES_REQUESTED** [review]
Please remove unwanted reports

File: Performance Script/WebhookLoadTestLoadTestWith3rps.js (hunk 13 above)
Comment (MitaliSingh16) [reply Performance Script/WebhookLoadTestLoadTestWith3rps.js:79]: done

File: Performance Script/WebhookLoadTestLoadTestWith3rps.js (hunk 12 above)
Comment (MitaliSingh16) [reply Performance Script/WebhookLoadTestLoadTestWith3rps.js:65]: done

File: Performance Script/WebhookLoadTestLoadTestWith3rps.js (hunk 11 above)
Comment (MitaliSingh16) [reply Performance Script/WebhookLoadTestLoadTestWith3rps.js:55]: done

**Review by Sadha1501493 — APPROVED** [review]
LGTM

**Review by Vishal14583 — APPROVED** [review]
LGTM

**Review by jrippington — APPROVED** [review]
Looks good

**Pull request merged**